sdk/src/main/java/io/opentelemetry/sdk/trace/SpanData.java

```diff
@@ -291,6 +315,14 @@ public SpanData build() {
       setAttributes(Collections.unmodifiableMap(new HashMap<>(getAttributes())));
       setTimedEvents(Collections.unmodifiableList(new ArrayList<>(getTimedEvents())));
       setLinks(Collections.unmodifiableList(new ArrayList<>(getLinks())));
+
+      // Calculate latency from start & end if possible.
+      if (!getLatencyNanos().isPresent()
+          && getStartEpochNanos().isPresent()
+          && getEndEpochNanos().isPresent()
+          && getEndEpochNanos().get() >= getStartEpochNanos().get()) {
```

Comment:
this is really ugly block of code. Can we at least move it into it's own method?

Reply:
Actually, this is only required for tests. The best option from my perspective would be to remove the latency from SpanData.

Reply:
I am fine with removing latency from the SpanData. For the EndEpoch what is the behavior you want?